client/src/components/ContentNavigator/ContentDataProvider.ts

```diff
@@ -138,8 +148,36 @@ class ContentDataProvider
     dataTransfer.set(this.dragMimeTypes[0], dataTransferItem);
   }
 
+  public async provideDocumentDropEdits(
+    _document: TextDocument,
+    position: Position,
+    dataTransfer: DataTransfer,
+    token: CancellationToken,
+  ): Promise<DocumentDropEdit | undefined> {
+    const dataTransferItem = dataTransfer.get(this.dragMimeTypes[0]);
+    const contentItem =
+      dataTransferItem && JSON.parse(dataTransferItem.value)[0];
+    if (token.isCancellationRequested || !contentItem) {
+      return undefined;
+    }
+
+    const fileFolderPath = await this.model.getFileFolderPath(contentItem);
+    if (!fileFolderPath) {
+      return undefined;
+    }
+
+    const filename = (parse(contentItem.name).name || contentItem.name).replace(
+      /[^a-zA-Z0-9]/g,
+      "",
```

Comment:
Well, there's a problem with this. If the item name is pure Chinese, say `中文.sas`, it will result in no name.

Reply:
That's a good point. What would you recommend in that instance? We could always do something like...
```
`file${Date.now()}`
```

That said, `file1695040695541` isn't the most readable variable name.

Or, we could start with a variable name something like `csvfile1`, and increment until we get one that _isn't_ in the document already? Maybe I'll mess around with that approach a bit.

I'll also look into the capitalization bit you mentioned previously. How would you make that determination? Could we look at the document, check if it has uppercase words, and use uppercase in that instance. Otherwise, default to lowercase?

Reply:
I added a different approach and more tests, here - https://github.com/sassoftware/vscode-sas-extension/pull/510/commits/6715b3aed01dd84acb56838dca7a8d5244250864